cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -251,11 +317,15 @@ private void destroy(WorkflowActionSpecification actionSpec, WorkflowAction acti
     return builder.build();
   }
 
-  private Supplier<WorkflowStatus> createStatusSupplier() {
-    return new Supplier<WorkflowStatus>() {
+  private Supplier<List<WorkflowActionNode>> createStatusSupplier() {
+    return new Supplier<List<WorkflowActionNode>>() {
       @Override
-      public WorkflowStatus get() {
-        return workflowStatus;
+      public List<WorkflowActionNode> get() {
+        List<WorkflowActionNode> currentNodes = Lists.newArrayList();
+        for (Map.Entry<String, WorkflowActionNode> entry : status.entrySet()) {
```

Comment:
Make `status` map an instance of ConcurrentHashMap, otherwise you need to synchronize the status map (even it's a synchronizedMap, see javadoc).